networking/eth2/src/integration-test/java/tech/pegasys/teku/networking/eth2/PingIntegrationTest.java

```diff
@@ -114,4 +116,64 @@ public void testPingPeriodically() throws Exception {
     // detecting that ping was automatically sent via updated att subnets of the remote peer
     waitFor(() -> assertThat(peer1.getRemoteAttestationSubnets()).contains(expectedBitvector1));
   }
+
+  @Test
+  public void testManualPingTimeout() throws Exception {
+    // sending PING manually to check eth2RpcOutstandingPingThreshold works correctly
+    Duration pingPeriod = Duration.ofDays(1);
+    network1 =
+        networkFactory
+            .builder()
+            .eth2RpcPingInterval(pingPeriod)
+            .eth2RpcOutstandingPingThreshold(2)
+            // remove PING method
+            .rpcMethodsModifier(m -> Stream.of(m).filter(t -> !t.getId().contains("/ping")))
+            .startNetwork();
+    network2 =
+        networkFactory
+            .builder()
+            .eth2RpcPingInterval(pingPeriod)
+            .eth2RpcOutstandingPingThreshold(2)
+            .peer(network1)
+            .startNetwork();
+
+    peer1 = network2.getPeer(network1.getNodeId()).orElseThrow();
+    peer2 = network1.getPeer(network2.getNodeId()).orElseThrow();
+
+    assertThatThrownBy(() -> peer1.sendPing().get(5, TimeUnit.SECONDS)).isNotNull();
+    assertThat(peer1.isConnected()).isTrue();
+    assertThatThrownBy(() -> peer1.sendPing().get(5, TimeUnit.SECONDS)).isNotNull();
+    assertThat(peer1.isConnected()).isTrue();
+    // the 3rd ping attempt should disconnect the peer since there are 2 unanswered PING requests
+    assertThatThrownBy(() -> peer1.sendPing().get(5, TimeUnit.SECONDS)).isNotNull();
+    waitFor(() -> assertThat(peer1.isConnected()).isFalse());
+  }
+
+  @Test
+  public void testAutomaticPingTimeout() throws Exception {
+    // PING is send automatically each 100ms.
+    // The peer should be finally disconnected as it doesn't answer
+    Duration pingPeriod = Duration.ofMillis(100);
```

Comment:
I think this is too fast - looks like we're sometimes disconnecting the peer before the network notices we're connected.

Reply:
Yeah, tried to fix this in a 'right' way with overriding `BeaconChainMethods` in `Eth2PeerManager` but came up with a pretty long chain of changes which are subject for a separate PR. 

I can change this to 1000ms but it still can fail potentially. 
Or I can remove this test at all since it nearly duplicates the previous test

Reply:
yeah - I think the manual one is pretty good.  maybe we should just cut this one ...